StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Views/LinkInstantDebitMandateView.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  LinkInstantDebitMandateView.swift
+//  StripePaymentSheet
+//
+//  Created by Ramon Torres on 2/17/22.
+//  Copyright © 2022 Stripe, Inc. All rights reserved.
+//
+
+@_spi(STP) import StripePaymentsUI
+@_spi(STP) import StripeUICore
+import UIKit
+
+protocol LinkInstantDebitMandateViewDelegate: AnyObject {
+    /// Called when the user taps on a link.
+    ///
+    /// - Parameters:
+    ///   - mandateView: The view that the user interacted with.
+    ///   - url: URL of the link.
+    func instantDebitMandateView(_ mandateView: LinkInstantDebitMandateView, didTapOnLinkWithURL url: URL)
+}
+
+// TODO(ramont): extract common code with `LinkLegalTermsView`.
+
+/// For internal SDK use only
+@objc(STP_Internal_LinkInstantDebitMandateViewDelegate)
+final class LinkInstantDebitMandateView: UIView {
+    struct Constants {
+        static let lineHeight: CGFloat = 1.5
+    }
+
+    // TODO(ramont): Update with final URLs
+    private let links: [String: URL] = [
+        "terms": URL(string: "https://stripe.com/ach-payments/authorization")!
+    ]
+
+    weak var delegate: LinkInstantDebitMandateViewDelegate?
+
+    private lazy var textView: UITextView = {
+        let textView = UITextView()
+        textView.isScrollEnabled = false
+        textView.isEditable = false
+        textView.backgroundColor = .clear
+        textView.attributedText = formattedLegalText()
+        textView.textColor = .linkSecondaryText
+        textView.textAlignment = .center
+        textView.textContainerInset = .zero
+        textView.textContainer.lineFragmentPadding = 0
+        textView.delegate = self
+        textView.clipsToBounds = false
+        textView.adjustsFontForContentSizeCategory = true
+        textView.linkTextAttributes = [
+            .foregroundColor: UIColor.linkBrandDark
+        ]
+        textView.font = LinkUI.font(forTextStyle: .caption)
+        return textView
+    }()
+
+    init(delegate: LinkInstantDebitMandateViewDelegate? = nil) {
+        super.init(frame: .zero)
+        self.delegate = delegate
+        addAndPinSubview(textView)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func formattedLegalText() -> NSAttributedString {
+        let string = STPLocalizedString(
+            "By continuing, you agree to authorize payments pursuant to these <terms>terms</terms>.",
+            "Mandate text displayed when paying via Link instant debit."
+        )
+
+        let formattedString = NSMutableAttributedString()
+
+        STPStringUtils.parseRanges(from: string, withTags: Set<String>(links.keys)) { string, matches in
```

Comment:
We should be able to leverage:
STPStringUtils.applyLinksToString

At the very least add a todo, if you feel it is out of scope for this PR

Reply:
Good call